tests/kernel/usleep/testcase.yaml

```diff
@@ -0,0 +1,5 @@
+tests:
+  kernel.usleep:
+    tags: sleep
+    arch_whitelist: x86
```

Comment:
seems a restrictive whitelist/platform_exclude list.
Is there something which prevents this from working on other arches?

Reply:
The platform_exclude list comes from the x86 boards that don't have an HPET.  The local APIC timer driver isn't up-to-date and doesn't work properly in TICKLESS mode, which is why they're excluded.  (I'll be addressing the local APIC timer driver problems, but that's a separate matter.)

As for the arch restriction - there's some discussion in #16075 about why these sleeps are a bit touch-and-go on various platforms.  For example, the test, as written, will not work on those platforms (the nRFs?) with a 32.768kHz timer source, since test selects a tick of 10us, which is below the hardware timer's resolution.  (I got that number -- 10us -- from a requirement in a project in my lap, so it's not entirely arbitrary.)

In order for the test to be meaningful, we should be testing platforms near the limits of their hardware timers' resolutions, but of course that means tuning the test for various platforms in the CI.  The [admittedly] lazy part of me thinks that's entirely too much work when all *I* need is for this to work right now on x86 with an HPET, and after all, I wrote in the docs that this API is dodgy, right?  On the other hand, as you are rightly pointing out, that's probably being a bit *too* lazy.  I could use some help here figuring out the right balance between the need for good test coverage and the investment of time required...

Reply:
I've rewritten the test to be agnostic about absolute values, so I've removed the restrictions on the test.